pkgs/top-level/all-packages.nix

```diff
@@ -13765,8 +13765,21 @@ with pkgs;
 
   pipenv = callPackage ../development/tools/pipenv {};
 
-  pipewire = callPackage ../development/libraries/pipewire {};
+  pipewire =
+    let
+      # Break a dependency cycle
+      customSDL2 = SDL2.override {
+        pipewireSupport = false;
+      };
+    in callPackage ../development/libraries/pipewire {
+      SDL2 = customSDL2;
```

Comment:
What does pipewire need SDL for anyways?

What's the closure size effect on this?

Reply:
I'm *pretty sure* Pipewire just uses SDL2 for examples.

Reply:
Not sure, but It's in `buildInputs` and feeding it regular SDL2 causes infinite recursion because of SDL2's pipewireSupport.

Reply:
What do you mean by examples?

Reply:
Things not required for the library itself.
https://github.com/PipeWire/pipewire/tree/0.3.44/src/examples

I think we should disable building those instead https://github.com/PipeWire/pipewire/blob/0.3.44/meson.build#L331

Reply:
Not quite sure how I'd go with it, seems like it does want SDL2 in `meson.build`, maybe I need to use some patches?

Reply:
Perhaps there's a flag to disable the examples?

Reply:
All the code that uses SDL2 in pipewire appears indeed to be some video examples. And there is an auto-option in meson_options.txt in pipewire, so it should be safe to simply drop SDL2 from the pipewire build inputs. Feel free to do that in this PR.

Reply:
SDL2 is required only for examples, however, it needs to be turned off manually. Somehow, the build fails if no SDL2 is present.

Here's how to do it:

``` patch
diff --git a/pkgs/development/libraries/pipewire/default.nix b/pkgs/development/libraries/pipewire/default.nix
index 79682d7ace8..640b002d29b 100644
--- a/pkgs/development/libraries/pipewire/default.nix
+++ b/pkgs/development/libraries/pipewire/default.nix
@@ -20,7 +20,6 @@
 , udev
 , libva
 , libsndfile
-, SDL2
 , vulkan-headers
 , vulkan-loader
 , webrtc-audio-processing
@@ -129,7 +128,6 @@ let
       vulkan-headers
       vulkan-loader
       webrtc-audio-processing
-      SDL2
       systemd
     ] ++ lib.optionals gstreamerSupport [ gst_all_1.gst-plugins-base gst_all_1.gstreamer ]
     ++ lib.optionals libcameraSupport [ libcamera libdrm ]
@@ -169,6 +167,7 @@ let
       "-Dsession-managers="
       "-Dvulkan=enabled"
       "-Dx11=${mesonEnableFeature x11Support}"
+      "-Dsdl2=disabled" # required only to build examples, causes dependency loop
     ];
 
     # Fontconfig error: Cannot load default config file
```